app/src/main/java/com/keylesspalace/tusky/json/GuardedAdapter.kt

```diff
@@ -35,10 +35,11 @@ class GuardedAdapter<T> private constructor(
 
     override fun fromJson(reader: JsonReader): T? {
         return try {
-            delegate.fromJson(reader)
+            delegate.fromJson(reader.peekJson())
         } catch (e: JsonDataException) {
```

Comment:
You should probably catch all kinds of exceptions, not just `JsonDataException`.
`JsonDataException` is what Moshi itself and its official adapters throw but you may have other unhandled exceptions like `IllegalArgumentException` in your own custom adapters.